src/python/pants/backend/scala/dependency_inference/scala_parser_test.py

```diff
@@ -468,3 +472,27 @@ class Class {
         "foo.valAnnotation",
         "foo.varAnnotation",
     ]
+
+
+def test_recursive_objects(rule_runner: RuleRunner) -> None:
```

Comment:
Should there also be a test of the objects being in different files? Maybe a test akin to the example from the PR description?

Reply:
I will add the test from the description as well. Should it be in the scalac_test? as it is going to invoke scalac

Reply:
You don't need to useaScalac per se. You can just ask the engine to run dependency inference. Just request `InferredDependencies` with `InferScalaSourceDependencies` as the request.

Reply:
Just pushed a test with both scenarios, calling both a method from the super and concrete types.
I run the test before and after the recursive change, the test only succeeds after the change.